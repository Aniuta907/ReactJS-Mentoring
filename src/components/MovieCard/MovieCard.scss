.movie {
  width: 300px;
  height: 540px;
  padding-right: 50px;

  &-text {
    text-transform: capitalize;
    color: #ffffff;
    opacity: 0.5;
  }

  &-wrapper-year {
    border: 1px solid #555555;
    width: 50px;
    height: 24px;
    border-radius: 3px;
  }
  
  &-year {
    vertical-align: middle;
  }

  &-title {
    font-size: large;
    margin-bottom: 10px;
    text-align: left;
  }

  &-genre {
    font-size: medium;
    text-align: left;
  }

  &-image {
    width: 300px;
    height: 420px;
    background-color: #555555;
    margin-bottom: 25px;
  }

  &-footer {
    display: flex;
    justify-content: space-between;
  }

  &-wrapper {
    display: flex;
    flex-direction: column;
  }

  &-dropdown {
    position: relative;
    top: 12px;
    right: 12px;
  }
}
