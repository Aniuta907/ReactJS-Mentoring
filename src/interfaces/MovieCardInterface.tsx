export interface MovieCardInterface {
    id: string;
    title: string;
    genres: Array<string>;
    release_date: string;
    poster_path: string;
    vote_average: string;
    overview: string;
    runtime: string;
}
  