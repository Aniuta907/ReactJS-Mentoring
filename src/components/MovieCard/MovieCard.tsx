import React from "react";

import { DottedIcon } from "../DottedIcon";
import "./MovieCard.scss";
interface MovieCardProps {
  releaseDate: string;
  title: string;
  genres: Array<string>;
  posterPath: string;
  id: string;
  onClick: () => void;
}

export const MovieCard: React.FC<MovieCardProps> = ({
  releaseDate,
  title,
  genres,
  posterPath,
  id,
  onClick,
}) => {
  return (
    <div className="movie">
      <div className="movie-dropdown">
        <DottedIcon id={id} />
      </div>
      <img className="movie-image" src={posterPath} onClick={onClick} />
      <div className="movie-footer">
        <div className="movie-wrapper">
          <span className="movie-title movie-text">{title}</span>
          <span className="movie-genre movie-text">
            {genres.length === 2 ? genres.join(" & ") : genres.join(", ")}
          </span>
        </div>
        <div className="movie-wrapper-year">
          <span className="movie-year movie-text">{releaseDate.split("-")[0]}</span>
        </div>
      </div>
    </div>
  );
};
