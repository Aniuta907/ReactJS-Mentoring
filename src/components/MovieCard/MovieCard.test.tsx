import * as React from "react";
import { shallow, ShallowWrapper } from 'enzyme';
import { createMock } from 'ts-auto-mock';

import { MovieCard } from './MovieCard';
import { MovieCardInterface } from '../../interfaces/MovieCardInterface';

const movieMock: MovieCardInterface = createMock<MovieCardInterface>();

const props = {
    movie: movieMock,
};

describe('MovieCard', () => {
    it('should render correctly', () => {
        const wrapper: ShallowWrapper = shallow(<MovieCard {...props} />);
        expect(wrapper).toMatchSnapshot();
    });
});
