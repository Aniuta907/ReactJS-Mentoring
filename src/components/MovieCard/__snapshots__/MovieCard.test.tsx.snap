// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MovieCard should render correctly 1`] = `
<div
  className="movie"
>
  <div
    className="movie-dropdown"
  >
    <DottedIcon
      id=""
    />
  </div>
  <Link
    to="/movies/"
  >
    <img
      className="movie-image"
      src=""
    />
  </Link>
  <div
    className="movie-footer"
  >
    <div
      className="movie-wrapper"
    >
      <span
        className="movie-title movie-text"
      />
      <span
        className="movie-genre movie-text"
      />
    </div>
    <div
      className="movie-wrapper-year"
    >
      <span
        className="movie-year movie-text"
      />
    </div>
  </div>
</div>
`;
